chrome/content/zotero/xpcom/attachments.js

```diff
@@ -2449,13 +2593,14 @@ Zotero.Attachments = new function(){
 			attachmentItem.attachmentLinkMode = linkMode;
 			attachmentItem.attachmentContentType = contentType;
 			attachmentItem.attachmentCharset = charset;
-			if (file) {
-				attachmentItem.attachmentPath = typeof file == 'string' ? file : file.path;
-			}
 			
 			if (collections) {
 				attachmentItem.setCollections(collections);
 			}
+
+			if (file) {
```

Comment:
Since `setCollections()` sets a default `libraryID` if one isn't already set, this re-ordering is just to give it one more chance to be set before possibly using it in `.attachmentPath`.